### PR TITLE
Dark themes

### PR DESCRIPTION
**Pull request by agilob**
See commits.

This is unfinished, because I don't know how to style chat area, there are colors, styles and sizes hardocded in a format of Qt::white instead of label->setStyleSheet("QLabel{color:white;}");

qtox-darkblue
qtox-wombat

 with adjustments from @kushagrasurana
qtox-light1

qtox-darkblue_chat

@krepa098 Could you help me with chat area styling? I removed some css'es to apply global theme on the header and message box, but I have no clue how to style messages. I found Text class which has hardcoded bolding, colors and size. I don't know how to use global theme on those classes.

### CODE REVIEW

**Comment by zetok** [comment]
why use `.png` rather than `.svg` ?

**Comment by agilob** [comment]
Don't have svg.

**Comment by tux3** [comment]
It fails to build for me.

```
../../projects/qtox/src/widget/form/settings/generalform.cpp:99:17: error: no member named 'styleColorCBox' in 'Ui::GeneralSettings'; did you mean 'themeColorCBox'?
        bodyUI->styleColorCBox->addItem(color);
                ^~~~~~~~~~~~~~
                themeColorCBox
../../projects/qtox/ui_generalsettings.h:89:16: note: 'themeColorCBox' declared here
    QComboBox *themeColorCBox;
               ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:100:13: error: no member named 'styleColorCBox' in 'Ui::GeneralSettings'; did you mean 'themeColorCBox'?
    bodyUI->styleColorCBox->setCurrentIndex(Settings::getInstance().getThemeColor());
            ^~~~~~~~~~~~~~
            themeColorCBox
../../projects/qtox/ui_generalsettings.h:89:16: note: 'themeColorCBox' declared here
    QComboBox *themeColorCBox;
               ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:103:13: error: no member named 'themeComboBox' in 'Ui::GeneralSettings'
    bodyUI->themeComboBox->addItems(Settings::getInstance().getThemesAvailable());
    ~~~~~~  ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:104:13: error: no member named 'themeComboBox' in 'Ui::GeneralSettings'
    bodyUI->themeComboBox->setCurrentText(Settings::getInstance().getTheme()); //current theme
    ~~~~~~  ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:165:21: error: no member named 'styleColorCBox' in 'Ui::GeneralSettings'; did you mean 'themeColorCBox'?
    connect(bodyUI->styleColorCBox, SIGNAL(currentIndexChanged(int)), this, SLOT(onStyleColorChanged(int)));
                    ^~~~~~~~~~~~~~
                    themeColorCBox
../../projects/qtox/ui_generalsettings.h:89:16: note: 'themeColorCBox' declared here
    QComboBox *themeColorCBox;
               ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:166:21: error: no member named 'themeComboBox' in 'Ui::GeneralSettings'
    connect(bodyUI->themeComboBox, &QComboBox::currentTextChanged, this, &GeneralForm::onThemeChanged);
            ~~~~~~  ^
../../projects/qtox/src/widget/form/settings/generalform.cpp:432:25: error: no member named 'styleColorCBox' in 'Ui::GeneralSettings'; did you mean 'themeColorCBox'?
    int index = bodyUI->styleColorCBox->currentIndex();
                        ^~~~~~~~~~~~~~
                        themeColorCBox
../../projects/qtox/ui_generalsettings.h:89:16: note: 'themeColorCBox' declared here
    QComboBox *themeColorCBox;
```

**Comment by agilob** [comment]
make clean 
make

**Comment by tux3** [comment]
@agilob Indeed, sorry about that.

**Comment by ghost** [comment]
SVGs can be made. I dont think it is wise adding more PNGs after all the effort i put in to get rid of them. HDPI displays getting more and more popular - pngs will look like pile of poo on them. I dont think we want that.

Otherwise i would love having dark theme, great job!

**Comment by agilob** [comment]
@r-ku could you make those SVGs and push them to my branch?

**Comment by ghost** [comment]
Im not really arts person sadly..

**Comment by zetok** [comment]
> @r-ku could you make those SVGs and push them to my branch?

That would still keep those PNGs though..

**Comment by Zer0-One** [comment]
I can't draw an SVG myself (I'm no artist), but if you want, I could attempt an automated trace of the PNGs. I've done it before, and it can work well, but I'd need much higher resolution images than what you include here.

File: src/widget/form/settings/generalform.cpp [hunk 1]
```diff
@@ -96,10 +96,13 @@ GeneralForm::GeneralForm(SettingsWidget *myParent) :
         bodyUI->styleBrowser->setCurrentText(tr("None"));
 
     for (QString color : Style::themeColorNames)
-        bodyUI->themeColorCBox->addItem(color);
-
-    bodyUI->themeColorCBox->setCurrentIndex(Settings::getInstance().getThemeColor());
+        bodyUI->styleColorCBox->addItem(color);
+    bodyUI->styleColorCBox->setCurrentIndex(Settings::getInstance().getThemeColor());
```
Comment (zetok) [inline src/widget/form/settings/generalform.cpp:100]: coding style nitpicking:
after

```
for ()
    something;
```

there should be a newline

Comment (agilob) [inline src/widget/form/settings/generalform.cpp]: don't care about it.

patches are welcome.

**Comment by agilob** [comment]
This branch is completely useless not because of +100 conflicts it makes when pulling upstream code. Closing, for reference use https://github.com/tux3/qTox/pull/1882 since now.

**Comment by agilob** [comment]
@r-ku  @Zer0-One If you could provide SVG instead of those .pngs I would be great!

**Pull request closed without merging**